packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -125,9 +129,25 @@ contract BridgePool is Testable {
         uint64 proposerRewardPct
     ) public {
         require(realizedFeePct <= maxFeePct, "Invalid realized fee");
-        Deposit storage newDeposit = deposits[depositId];
+
+        // Construct unique ancillary data for this deposit, and link relay data to it.
+        RelayAncillaryDataContents memory relayData =
+            RelayAncillaryDataContents({
+                depositId: depositId,
+                l2Sender: l2Sender,
+                recipient: recipient,
+                depositTimestamp: depositTimestamp,
+                l1Token: l1Token,
+                amount: amount,
+                maxFeePct: maxFeePct,
+                proposerRewardPct: proposerRewardPct,
+                realizedFeePct: realizedFeePct,
+                slowRelayer: msg.sender
+            });
+        bytes memory priceRequestAncillaryData = getRelayAncillaryData(relayData);
```

Comment:
What do you think the pros/cons are of passing in the hash in as the only param in ancillary data and asking the voters to look up the hash to determine the validity of the data?

Reply:
I think this is a reasonable feature, just adds more client complexity in exchange for gas efficiency. @chrismaree wdyt?

Reply:
ye, I agree the hash is better. the OO callback on `priceDisputed` returns the ancillary data from the price request which you can hash to get the key for this on the dispute side. the event emitted called `DepositRelayed` contains everything to compute the hash so no loss or big complexity I don't think.